docker-app/qfieldcloud/core/rest_utils.py

```diff
@@ -18,14 +19,18 @@ def exception_handler(exc, context):
         qfc_exc = qfieldcloud_exceptions.NotAuthenticatedError()
     elif isinstance(exc, rest_exceptions.PermissionDenied):
         qfc_exc = qfieldcloud_exceptions.PermissionDeniedError()
-    elif isinstance(exc, exceptions.ObjectDoesNotExist):
+    elif isinstance(exc, (exceptions.ObjectDoesNotExist, Http404)):
         qfc_exc = qfieldcloud_exceptions.ObjectNotFoundError(detail=str(exc))
     elif isinstance(exc, exceptions.ValidationError):
         qfc_exc = qfieldcloud_exceptions.ValidationError(detail=str(exc))
     elif isinstance(exc, qfieldcloud_exceptions.QFieldCloudException):
         qfc_exc = exc
     elif isinstance(exc, rest_exceptions.APIException):
-        qfc_exc = qfieldcloud_exceptions.APIError(exc.detail, exc.status_code)
+        # Map DRF API exceptions to qfc exceptions. Note the `detail` attribute is always present but not properly typed in `APIException`
+        qfc_exc = qfieldcloud_exceptions.APIError(
+            detail=getattr(exc, "detail"),
```

Comment:
Make the handler more robust in case of a custom exception (inherited from APIException) does not include `detail`: 

```suggestion
            detail=getattr(exc, "detail", str(exc))
```

Reply:
Actually I tried (and seems I failed) to bring the atttention at the fact that `details` is always present and I did this change only to make the typing happy. We must throw if there is no `detail`, because something went very wrong.

Reply:
Fair enough , thanks for the clarification!

Reply:
Just pushed a new commit here, hopefully now the comment says it all.